data_collection/gazette/spiders/base/brtransparencia.py

```diff
@@ -0,0 +1,68 @@
+import re
+from datetime import datetime
+
+from scrapy import Request
+from scrapy.selector import Selector
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class BaseBrTransparenciaSpider(BaseGazetteSpider):
+    name = ""
+    TERRITORY_ID = ""
+    allowed_domains = []
+    start_urls = [""]
+    power = "executive"
+
+    def _extract_code_from_response_text(self, response_text, field="entity"):
+        return re.search(
+            rf'var {field}(\ )*=(\ )*["|\'](.+?)["|\']',
+            response_text,
+            re.IGNORECASE,
+        ).groups()[2]
+
+    def _extract_entity_code(self, response):
+        response_text = response.text
+        try:
+            response_entity = self._extract_code_from_response_text(
+                response_text, field="entity"
+            )
+        except AttributeError as exc:
+            raise AttributeError("Was not possible to extract the entity code") from exc
+        try:
+            response_code = self._extract_code_from_response_text(
+                response_text, field="code"
+            )
+        except AttributeError as exc:
+            raise AttributeError("Was not possible to extract the code") from exc
```

Comment:
@talesmota nesses casos de exceção , é preferivél que deixe o raspador quebrar direto apenas, ou se preferir adicionar no log de erro e manter a exceção original
```suggestion
        except AttributeError:
            self.logger.error("Was not possible to extract the entity code")
            raise
        try:
            response_code = self._extract_code_from_response_text(
                response_text, field="code"
            )
        except AttributeError:
            self.logger.error("Was not possible to extract the code")
            raise
```